devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy

```diff
@@ -0,0 +1,159 @@
+		standardTile("lock", "device.lock", inactiveLabel:false, decoration:"flat", width:2, height:2) {
+			state "default", label:'lock', action:"lock.lock", icon:"st.locks.lock.locked", nextState:"locking"
+		}
+		standardTile("unlock", "device.lock", inactiveLabel:false, decoration:"flat", width:2, height:2) {
+			state "default", label:'unlock', action:"lock.unlock", icon:"st.locks.lock.unlocked", nextState:"unlocking"
+		}
+		valueTile("battery", "device.battery", inactiveLabel:false, decoration:"flat", width:2, height:2) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+		standardTile("refresh", "device.lock", inactiveLabel:false, decoration:"flat", width:2, height:2) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+
+		main "toggle"
+		details(["toggle", "lock", "unlock", "battery", "refresh"])
+	}
+}
+
+// Globals
+private getCLUSTER_POWER() { 0x0001 }
+private getCLUSTER_DOORLOCK() { 0x0101 }
+
+private getDOORLOCK_CMD_LOCK_DOOR() { 0x00 }
+private getDOORLOCK_CMD_UNLOCK_DOOR() { 0x01 }
+private getDOORLOCK_ATTR_LOCKSTATE() { 0x0000 }
+private getPOWER_ATTR_BATTERY_PERCENTAGE_REMAINING() { 0x0021 }
+
+// Public methods
+def installed() {
+    log.trace "installed()"
+}
+
+def uninstalled() {
+    log.trace "uninstalled()"
+}
+
+def configure() {
+    def cmds =
+        zigbee.configSetup("${CLUSTER_DOORLOCK}", "${DOORLOCK_ATTR_LOCKSTATE}",
+                           "0x30", 0, 3600, "{01}") +
+        zigbee.configSetup("${CLUSTER_POWER}", "${POWER_ATTR_BATTERY_PERCENTAGE_REMAINING}",
+                           "0x20", 3600, 3600, "{01}")
+    log.info "configure() --- cmds: $cmds"
+    return cmds + refresh() // send refresh cmds as part of config
+}
+
+def refresh() {
+    def cmds =
+        zigbee.refreshData("${CLUSTER_DOORLOCK}", "${DOORLOCK_ATTR_LOCKSTATE}") +
+        zigbee.refreshData("${CLUSTER_POWER}", "${POWER_ATTR_BATTERY_PERCENTAGE_REMAINING}")
+    log.info "refresh() --- cmds: $cmds"
+    return cmds
+}
+
+def parse(String description) {
+    log.trace "parse() --- description: $description"
+
+    Map map = [:]
+    if (description?.startsWith('read attr -')) {
+        map = parseReportAttributeMessage(description)
+    }
+
+    log.debug "parse() --- Parse returned $map"
+    def result = map ? createEvent(map) : null
+    return result
+}
+
+// Lock capability commands
+def lock() {
+    def cmds = zigbee.zigbeeCommand("${CLUSTER_DOORLOCK}", "${DOORLOCK_CMD_LOCK_DOOR}", "{}")
+    log.info "lock() -- cmds: $cmds"
+    return cmds
+}
+
+def unlock() {
+    def cmds = zigbee.zigbeeCommand("${CLUSTER_DOORLOCK}", "${DOORLOCK_CMD_UNLOCK_DOOR}", "{}")
+    log.info "unlock() -- cmds: $cmds"
+    return cmds
+}
+
+// Private methods
+private Map parseReportAttributeMessage(String description) {
+    log.trace "parseReportAttributeMessage() --- description: $description"
+
+    Map descMap = (description - "read attr - ").split(",").inject([:]) { map, param ->
```

Comment:
Instead of this closure here, you can use the zigbee.parseDescriptionAsMap which does the name thing